Add Robot 101 post on The Code (#510)
srobo/website: _posts/2023-06-29-101-code.md

can identify potential issues and generate creative solutions that a single mind
might overlook. Think of it as having a built-in code review process!

## The Pulse of Your Robot - The Control Loop

Your robot interacts with the environment in a continuous cycle of sensing and
acting. It's essential to have a solid control loop in your code where your
robot senses its environment and then decides what to do next based on that
information. This loop should be at the core of your robot's operation, allowing
it to adapt and react to its surroundings.

## Backup Your Code: The Magic of Version Control Systems

Mistakes and accidents happen, files get lost, and computers crash. To protect
your work from these unpredictable events, ensure you regularly back up your
code. Better yet, use a Version Control System (VCS) like Git. A VCS not only
provides a backup of your code but also tracks changes, making it easy to
identify when and where things might have gone wrong or right!

## The Six T's: Test, Test, Test, Test, Test, Test!

If there's one golden rule in the world of coding, it's this: test your code.
Then, test it again, and keep testing! You can utilise our web interface to help

> you quickly iterate on your robot, by live streaming the logs and restarting

Reviewer: Should we also mention webcam images as well as logs?

Author: [f5c2794](https://github.com/srobo/website/pull/510/commits/f5c2794060f7aec8adbcc26ea8cd580ecc65d091)

> you quickly iterate on your robot, by live streaming the logs and restarting
> execution of your code. The more robust your testing, the more reliable your
> robot will be when it counts.

Author: ```suggestion
you quickly iterate on your robot, by live streaming the logs and viewing
the last captured image. You can also use the web interface to stop and 
restart the execution of your code. The more robust your testing, the 
more reliable your robot will be when it counts.
```

the brains of your robot - its code.

## Teamwork Makes The Code Work

First and foremost, your robot's code should never be a one-person show. Make
sure more than one person has access to the codebase and understands it. This is
crucial for two reasons: First, if the primary coder is unavailable for any
reason, others can step in and keep things moving. Second, multiple perspectives
can identify potential issues and generate creative solutions that a single mind
might overlook. Think of it as having a built-in code review process!

## The Pulse of Your Robot - The Control Loop

Your robot interacts with the environment in a continuous cycle of sensing and
acting. It's essential to have a solid control loop in your code where your
robot senses its environment and then decides what to do next based on that
information. This loop should be at the core of your robot's operation, allowing
it to adapt and react to its surroundings.

## Backup Your Code: The Magic of Version Control Systems

Mistakes and accidents happen, files get lost, and computers crash. To protect
your work from these unpredictable events, ensure you regularly back up your

> code. Better yet, use a Version Control System (VCS) like Git. A VCS not only

Reviewer 2: Is a specific mention of github itself worth it? I'd only heard of github and not git specifically before SR

Author: ```suggestion
code. Better yet, use a Version Control System (VCS) like Git with GitHub. A VCS not only
```

Author: ```suggestion
code. Better yet, use a Version Control System (VCS) like Git with GitHub. A VCS not only
```

acting. It's essential to have a solid control loop in your code where your
robot senses its environment and then decides what to do next based on that
information. This loop should be at the core of your robot's operation, allowing
it to adapt and react to its surroundings.

## Backup Your Code: The Magic of Version Control Systems

Mistakes and accidents happen, files get lost, and computers crash. To protect
your work from these unpredictable events, ensure you regularly back up your
code. Better yet, use a Version Control System (VCS) like Git with GitHub. A VCS not only
provides a backup of your code but also tracks changes, making it easy to
identify when and where things might have gone wrong or right!

## The Six T's: Test, Test, Test, Test, Test, Test!

If there's one golden rule in the world of coding, it's this: test your code.
Then, test it again, and keep testing! You can utilise our web interface to help
you quickly iterate on your robot, by live streaming the logs and viewing
the last captured image. You can also use the web interface to stop and 
restart the execution of your code. The more robust your testing, the 

> more reliable your robot will be when it counts.

Author: ```suggestion
more reliable your robot will be when it counts!
```

## The Benefits of Friendly Competition

If you have a large enough team, consider developing competing implementations
of certain features or systems. This can be an effective way to explore
different solutions and spark innovation. You can then choose the best elements
from each, or select the most efficient solution overall.

## Embrace Imperfections: Sensor Noise

> It's crucial to remember that real-world data are often noisy and imperfect.

Author: ```suggestion
It's crucial to remember that real-world data is often noisy and imperfect.
```

Your sensors may pick up interference, or there could be slight variations in
readings. Your code should be robust enough to handle this noise and still make

> effective decisions. Techniques such as sensor fusion or filtering can be highly
> beneficial here.

Author: ```suggestion
effective decisions. Techniques such as sensor fusion or filtering (like 
taking an average of multiple readings) can be highly beneficial here.
```

## The Power of Documentation

Our [Docs](https://studentrobotics.org/docs) are a goldmine of knowledge and 
should be your first port of call whenever you're unsure about something. Much 
like a good textbook or reference guide, our Docs are designed to support you in

> understanding your robot better. From getting started with your robot kit, to 
> understanding the APIs, to troubleshooting common issues, the Docs cover a broad
> spectrum of topics you may encounter.

Author: ```suggestion
understanding your robot better. From getting started with your robot kit, to 
understanding the APIs, to troubleshooting common issues; the Docs cover a broad
spectrum of topics you may encounter.
```

<br/>

> Coding your robot can be a challenging yet rewarding experience. It's where the
> magic happens, where the inert pieces of metal and circuits come to life and
> engage with the world. We hope these tips and strategies will help you avoid

Author: ```suggestion
Coding your robot can be a challenging yet rewarding experience. It's where the
magic happens, where the inert pieces of metal, wood, and circuitry come to life
and engage with the world. We hope these tips and strategies will help you avoid
```
